src/org/opendatakit/briefcase/ui/pull/PullPanel.java

```diff
@@ -75,17 +78,28 @@ public PullPanel(TransferPanelForm view, TransferForms forms, BriefcasePreferenc
         updateActionButtons();
       } catch (HttpException e) {
         log.warn("Unable to get form list from {}: {}", source.getDescription(), e.toString());
+        new SwingWorker() {
```

Comment:
It’s not clear to me why you’re using another thread here. What happens when you simply call showMessageDialog?  Perhaps a comment? The commit message, `- On startup, we need to launch it into a background thread to avoid blocking the UI.` suggests (to me) you’re doing the `getFormList` in another thread, but you aren’t.

Reply:
Thanks! I'll add a comment.

Swing's threading model is kind of broken and if I don't do that it will show the dialog before rendering the main frame, which is really confusing.

This way, the main frame gets rendered and the error dialog shows on top.